src/components/Teams/Teams.jsx

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState } from "react";
+import TeamMember from "../TeamMember/TeamMember";
+
+const Teams = () => {
+  const [isLoadedTeamsData, setIsLoadedTeamsData] = useState(false);
+  const [teamsData, setTeamsData] = useState([]);
+  const [teamMembersData, setTeamMembersData] = useState([]);
+  const [teamsError, setTeamsError] = useState(null);
+
+  useEffect(() => {
+  
+    //TODO: Add abort controller when fetching from API
```

Comment:
In case we want to abort the request to API if the component reloads without the previous request being fulfilled. It is not a necessity though.

Reply:
You mean so re-rendering the component doesn't request the data twice?

To solve this problem, you can have two bits of information:
1. A flag telling if a request is in progress.
2. The data itself, it can be an object or nullish.

Then the `useEffect` can look like this:
1. Check if there is a request in progress and just `return`.
2. Check if the data exists (i.e. is not `null`/`undefined`) and just `return`.
3. Add the logic for the data request itself, taking care to assign both the data and the flag.

This way even if the component re-renders it will not send multiple requests. 😉